### PR TITLE
network: refuse IPv4 multipath route for IPv6 route

### PR DESCRIPTION
**Pull request by yuwata**
Split out of #18589.

### CODE REVIEW

File: src/network/networkd-route.c [hunk 1]
```diff
@@ -2573,6 +2573,17 @@ static int route_section_verify(Route *route, Network *network) {
                 route->gateway_onlink = true;
         }
 
+        if (route->family == AF_INET6) {
```
Comment (bluca) [inline src/network/networkd-route.c:2576]: is the other way around possible and worth checking as well? INET route configured with INET6 multipath route

Comment (yuwata) [reply src/network/networkd-route.c:2576]: Good catch. But interestingly, IPv4 route with IPv6 multipath route is certainly supported by kernel and ip tools.

Comment (bluca) [reply src/network/networkd-route.c:2576]: oh, interesting

**Review by bluca — APPROVED** [review]
LGTM

**Pull request merged**